buildenv/jenkins/common/build.groovy

```diff
@@ -585,26 +585,75 @@ def cleanWorkspace (KEEP_WORKSPACE) {
     }
 }
 
+def clean_docker_containers() {
+    println("Listing docker containers to attempt removal")
+    sh "docker ps -a"
+    def containers = sh (script: "docker ps -a --format \"{{.ID}}\"", returnStdout: true).trim()
+    if (containers) {
+        println("Stop all docker containers")
+        sh "docker ps -a --format \"{{.ID}}\" | xargs docker stop"
+        println("Remove all docker containers")
+        sh "docker ps -a --format \"{{.ID}}\" | xargs docker rm --force"
+    }
+}
+
+def prepare_docker_environment() {
+    clean_docker_containers()
+    DOCKER_IMAGE_ID = get_docker_image_id(DOCKER_IMAGE)
+
+    // if there is no id found then attempt to pull it
+    if (!DOCKER_IMAGE_ID) {
+        echo "${DOCKER_IMAGE} not found locally, attempting to pull from dockerhub"
+        docker.withRegistry("", "9f50c848-8764-440d-b95a-1d295c21713e") {
+            sh "docker pull ${DOCKER_IMAGE}"
```

Comment:
Should we pull everytime? If a container is updated I believe it won't update on a machine unless we pull explicitly.

Reply:
The version number is specified in the defaults.yaml, in this case `jenkins-agent-x86-centos6:13`.  The pull is only required when the host system does not have that image local.